modules/security-jwt/src/main/java/com/devonfw/module/security/jwt/util/TokenExtractor.java

```diff
@@ -1,140 +1,104 @@
+ *
  */
-@Configuration
+@Named
 public class TokenExtractor {
 
-	private static final Logger LOG = LoggerFactory.getLogger(TokenExtractor.class);
-
-	private ObjectMapper objectMapper;
-
-	private KeyStoreAccess keyStoreAccess;
-
-	private Map<String, Object> tokenClaims;
-
-	public TokenExtractor(KeyStoreAccess keyStoreAccess,ObjectMapper objectMapper) {
-		this.keyStoreAccess = keyStoreAccess;
-		this.objectMapper=objectMapper;
-	}
-
-	public Authentication validateTokenAndSignature(String token) {
-
-		Authentication authentication = null;
-		// get algorithm from header
-		// create signatureverifier of particular type
-		// extract token
-		// deserialize it
-		// verify signature
-		// TODO:validate timestamp
-		// TODO: take data from token and create authentication
+  private static final Logger LOG = LoggerFactory.getLogger(TokenExtractor.class);
 
-		String algorithms = JwtHelper.headers(token).get("alg");
+  private ObjectMapper objectMapper = new ObjectMapper();
 
-		String algorithmFamilyType = getAlgorithmFamilyType(algorithms);
-		if (algorithmFamilyType == null) {
-			// throw exception
-		}
+  private Map<String, Object> tokenClaims;
 
-		SignatureVerifier verifier = signatureVerifierFactory(((RSAPublicKey)this.keyStoreAccess.getPublicKey()),
-				algorithmFamilyType);
+  @Inject
+  private JwtSignatureAlgorithmFactory jwtSignatureAlgorithmFactory;
 
-		Jwt jwt = JwtHelper.decodeAndVerify(token, verifier);
+  /**
+   * Validates the token
+   *
+   * @param token
+   *
+   * @return {@link Authentication}
+   */
+  public Authentication validateTokenAndSignature(String token) {
 
-		String claims = jwt.getClaims();
-		try {
-			tokenClaims = this.objectMapper.readValue(claims, Map.class);
+    Authentication authentication = null;
 
-			LOG.info("Token Claims " + tokenClaims.toString());
+    com.devonfw.module.security.jwt.sign.JwtSignatureAlgorithm jwtSignatureAlgorithm = this.jwtSignatureAlgorithmFactory
+        .getAlgorithms("RSA");
 
-			Instant now = Instant.now();
+    SignatureVerifier verifier = jwtSignatureAlgorithm.createVerifier();
+    Jwt jwt = JwtHelper.decodeAndVerify(token, verifier);
 
-			if(tokenClaims.get("nbf")!=null) {
-				Instant nbf = Instant.ofEpochSecond(((Number) tokenClaims.get("nbf")).longValue());
-				if (now.isBefore(nbf)) {
-					// Throw exception
-					LOG.info("Token is not valid before " + nbf);
-				}
+    String claims = jwt.getClaims();
+    try {
+      this.tokenClaims = this.objectMapper.readValue(claims, Map.class);
 
-			}
+      LOG.info("Token Claims " + this.tokenClaims.toString());
 
-			if(tokenClaims.get("exp") !=null) {
-				Instant expiration_time = Instant.ofEpochSecond(((Number) tokenClaims.get("exp")).longValue());
+      Instant now = Instant.now();
 
-				if (now.isAfter(expiration_time)) {
-					// Throw exception
-					LOG.info("Token is expired");
-				}
-			}
+      Object nbfObj = this.tokenClaims.get("nbf");
+      Object expObj = this.tokenClaims.get("exp");
```

Comment:
So here you are using hardcoded string literals for the claims? Why can they be configured then?
As I stated above define constants for that. Maybe the JWT API already provides such constants or even methods for the standardized claims.